apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -323,227 +323,270 @@ class ShareAllowedDialog extends React.Component {
 
     const warningText = this.getWarningText(showPublishInfo);
 
-    return (
-      <div>
-        <BaseDialog
-          style={styles.modal}
-          isOpen={isOpen}
-          handleClose={this.close}
-          hideBackdrop={hideBackdrop}
+    const interfaceVersion = appType === 'music' ? 2 : 1;
+
+    const renderAbuseError = () => {
+      return (
+        <AbuseError
+          i18n={{
+            tos: i18n.tosLong({url: 'http://code.org/tos'}),
+            contact_us: i18n.contactUs({
+              url: `https://support.code.org/hc/en-us/requests/new?&description=${encodeURIComponent(
+                `Abuse error for project at url: ${shareUrl}`
+              )}`,
+            }),
+          }}
+          className="alert-error"
+          style={styles.abuseStyle}
+          textStyle={styles.abuseTextStyle}
+        />
+      );
+    };
+
+    const renderSharingDisabled = () => {
+      return (
+        <div style={{position: 'relative'}}>
+          <div style={{paddingRight: 10}}>
+            <p>{i18n.sharingBlockedByTeacher()}</p>
+          </div>
+          <div style={{clear: 'both', height: 40}}>
+            <button
+              type="button"
+              id="continue-button"
+              style={{position: 'absolute', right: 0, bottom: 0, margin: 0}}
+              onClick={this.close}
+            >
+              {i18n.dialogOK()}
+            </button>
+          </div>
+        </div>
+      );
+    };
+
+    const renderCopyButton = () => {
+      return (
+        <Button
+          color={Button.ButtonColor.brandSecondaryDefault}
+          id="sharing-dialog-copy-button"
+          icon="clipboard"
+          style={{
+            ...styles.button,
+            ...styles.copyButton,
+            ...(this.state.hasBeenCopied && styles.copyButtonLight),
+          }}
+          onClick={wrapShareClick(
+            this.copy,
+            'SHARING_LINK_COPY',
+            this.props.appType
+          )}
+          text={i18n.copyLinkToProject()}
+          value={shareUrl}
+        />
+      );
+    };
+
+    const renderInterfaceV1 = () => {
```

Comment:
Not blocking, but could these just be a components? 

Maybe tweaked like this:

```
const AbuseErrorWrapper = ({ isAbusive }) => {
  if (!isAbusive) { return null}
  // return same as above
}

// and then later this:
{ isAbusive && renderAbuseError() }
// turns into this
<AbuseErrorWrapper {isAbusive={isAbusive} />
```

An argument could be made about handing in the `isAbusive` prop vs continuing to wrap it in the conditional. I'm also not completely convinced of what a good name for that wrapper component would be.

But in general, we should skew towards components instead of functions that return components.

Same argument and style for the other `render` functions. Obviously I wrote this up after just seeing the first one. 😇

Reply:
You're right.  I was trying to make minimal changes to the legacy code, but have built a revised approach which uses functional components.